server/app/services/program/ActiveAndDraftPrograms.java

```diff
@@ -70,30 +80,110 @@ private ImmutableMap<String, ProgramDefinition> mapNameToProgram(
     return mapNameToProgramWithFilter(repository, service, versionModel, Optional.empty());
   }
 
-  private ActiveAndDraftPrograms(VersionRepository repository, Optional<ProgramService> service) {
+  private ActiveAndDraftPrograms(
+      VersionRepository repository,
+      Optional<ProgramService> service,
+      ImmutableList<ActiveAndDraftProgramsType> types) {
```

Comment:
**suggestion:** Sorry for the back and forth, but I'm realizing [EnumSet](https://docs.oracle.com/javase/7/docs/api/java/util/EnumSet.html) would actually be better than a list here. It allows you to do things like EnumSet.allOf(ActiveAndDraftProgramsType) to indicate you want a set of all of them.

Reply:
No worries. I love the idea of using `EnumSet`, thank you so much for the suggestion! Changes applied.